sources/updater/updog/src/transport.rs

```diff
@@ -1,61 +1,78 @@
-use std::cell::{BorrowMutError, RefCell};
-use tough::{HttpTransport, Repository, Transport};
+use log::error;
+use std::sync::{Arc, RwLock};
+use tough::{HttpTransport, Transport, TransportError};
 use url::Url;
 
-#[derive(Debug)]
+/// A shared pointer to a list of query params that the transport will add to HTTP calls.
+#[derive(Debug, Clone, Default)]
+pub(crate) struct QueryParams(Arc<RwLock<Vec<(String, String)>>>);
+
+/// A `tough` `Transport` that allows us to add query parameters to HTTP calls.
+#[derive(Debug, Clone)]
 #[allow(clippy::module_name_repetitions)]
-pub struct HttpQueryTransport {
+pub(crate) struct HttpQueryTransport {
     pub inner: HttpTransport,
-    parameters: RefCell<Vec<(String, String)>>,
+    parameters: QueryParams,
 }
 
-impl HttpQueryTransport {
-    pub fn new() -> Self {
-        Self {
-            inner: HttpTransport::new(),
-            parameters: RefCell::new(vec![]),
+impl QueryParams {
+    pub(crate) fn add_params_to_url(&self, mut url: Url) -> Url {
+        let mut params = match self.0.write() {
+            Err(e) => {
+                // a thread died while holding a lock to the params. unlikely to occur.
+                error!("unable to add query params to HTTP call: {}", e);
+                return url;
+            }
+            Ok(lock_result) => lock_result,
+        };
+        params.sort_by(|(a, _), (b, _)| a.cmp(b));
+        for (key, val) in params.iter() {
+            url.query_pairs_mut().append_pair(&key, &val);
         }
```

Comment:
```suggestion
        url.query_pairs_mut().extend_pairs(params.iter());
```
Hopefully this works?